ci/test-checks.sh

```diff
@@ -44,6 +44,8 @@ echo --- build environment
 export RUST_BACKTRACE=1
 export RUSTFLAGS="-D warnings -A incomplete_features"
 
+_ "$cargo" stable clean
```

Comment:
is this intended to be included in this pr? i think this increase the build time sometimes needlessly by removing the build cache too much?

Reply:
It was my only way to get past a compiler issue on CI. I will remove before merging. I mentioned it in #devops on Discord as well